ambry-server/src/main/java/com.github.ambry.server/AmbryServer.java

```diff
@@ -158,7 +158,8 @@ public void startup() throws InstantiationException {
       networkServer = new SocketServer(networkConfig, sslConfig, registry, ports);
       requests =
           new AmbryRequests(storageManager, networkServer.getRequestResponseChannel(), clusterMap, nodeId, registry,
-              findTokenFactory, notificationSystem, replicationManager, storeKeyFactory);
+              findTokenFactory, notificationSystem, replicationManager, storeKeyFactory,
+              storeConfig.storeEnableDataPrefetch);
```

Comment:
The config itself is being read by the server, not the store as far as I can see. This probably should be a server config: `serverEnableStoreDataPrefetch`.